app/rest/db-logging/src/main/java/io/syndesis/rest/dblogging/controller/KubernetesSupport.java

```diff
@@ -0,0 +1,100 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.rest.dblogging.controller;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.internal.PodOperationsImpl;
+import io.fabric8.kubernetes.client.utils.HttpClientUtils;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * Provides some enriched operations against a KubernetesClient.
+ */
+public class KubernetesSupport {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KubernetesSupport.class);
+
+    private final KubernetesClient client;
+    private final OkHttpClient okHttpClient;
+
+    public KubernetesSupport(KubernetesClient client) {
+        this.client = client;
+        this.okHttpClient = this.client == null ? null : HttpClientUtils.createHttpClient(this.client.getConfiguration());
+    }
+
+
+    /*
+     * Feeds the controller of the given podName to the callback handler for processing.
+     *
+     * We do this instead of using the watchLog() feature of the k8s client lib because it really sucks due to:
+     *  1. You can't configure the timestamps option or the sinceTime option.  Need to resume log downloads.
+     *  2. It seems to need extra threads..
+     *  3. It might be hiding some of the http failure conditions.
+     *
+     */
+    protected void watchLog(String podName, Consumer<InputStream> handler, String sinceTime, Executor executor) throws IOException {
+        try {
+            PodOperationsImpl pod = (PodOperationsImpl) client.pods().withName(podName);
+            String url = pod.getResourceUrl().toString() + "/log?pretty=false&follow=true&timestamps=true";
+            if (sinceTime != null) {
+                url += "&sinceTime=" + URLEncoder.encode(sinceTime, "UTF-8");
+            }
+            // URL url = new URL(URLUtils.join());
+            Request request = new Request.Builder().url(new URL(url)).get().build();
+            OkHttpClient clone = okHttpClient.newBuilder().readTimeout(0, TimeUnit.MILLISECONDS).build();
+            clone.newCall(request).enqueue(new Callback() {
```

Comment:
How is a restart handled when a connection is dropped ?

Reply:
That should be handled by this catch:
https://github.com/chirino/syndesis/blob/028131b4abf79f7d30e000f8bd26969ebc1db1c9/app/rest/db-logging/src/main/java/io/syndesis/rest/dblogging/controller/PodLogHandler.java#L101

Basically the pod log handler handles any errors by re-scheduling a run of the handler until the handler is shutdown.